### ISSUE TITLE
CI: improve perf

### ISSUE THREAD

**Issue by Arqu** [n0-computer/iroh#1717]
From some humble weekend testing we can gain a lot of perf improvement with a good setup of sccache and a better linker.

Also try out nextest

**Comment by Arqu** [comment]
Windows is lacking in this regard. Will file a separate issue.

**Comment by Arqu** [comment]
FYI nextest is a great option for a future revision. It will allow us to split up test runs and do them in parallel across multiple instances. So we can have 10x 3.5 min jobs instead of 1x 30 min job. Obviously needs infra improvements to support all this but we really don't have much options if we want to keep runtime sub 10 min.

Tried out the mold linker, improvements in our case were marginal at best for CI. Or I held it wrong. Will do again on our next CI perf run.
